processing/src/main/java/org/apache/druid/query/monomorphicprocessing/SpecializationService.java

```diff
@@ -65,17 +69,158 @@
 {
   private static final Logger LOG = new Logger(SpecializationService.class);
 
-  private static final Unsafe UNSAFE;
+  private static final MethodHandle DEFINE_CLASS;
```

Comment:
Could you please extract "defining class" functionality in a separate class, a-la `ByteBufferUtils`? It's better to keep `SpecializationService` shorter and more focused.

Reply:
sure, happy to do that